### PR TITLE
Simplify and test FTS boundary conditions

### PR DESCRIPTION
**Pull request by simone-silvestri**
This PR simplifies a bit the code behind FTS bc and adds tests for them.
In addition it removes the option to index reduced FTS with only two spatial indices since that functionality does not exist for Field either.

### CODE REVIEW

File: src/OutputReaders/field_time_series_indexing.jl [hunk 1]
```diff
@@ -172,21 +172,14 @@ const YFTS  = FlavorOfFTS{Nothing, <:Any, Nothing}
...
-@propagate_inbounds getindex(f::YFTS,  i::Int, k::Int, n::Time) = getindex(f, 1, j, 1, n)
-@propagate_inbounds getindex(f::ZFTS,  j::Int, k::Int, n::Time) = getindex(f, 1, 1, k, n)
-@propagate_inbounds getindex(f::FTS0,  j::Int, k::Int, n::Time) = getindex(f, 1, 1, 1, n)
+# getbc for FTS boundary conditions
+@inline getbc(f::XYFTS, i::Int, j::Int, grid::AbstractGrid, clock, args...) = @inbounds f[i, j, 1, Time(clock.time)]
+@inline getbc(f::XZFTS, i::Int, k::Int, grid::AbstractGrid, clock, args...) = @inbounds f[i, 1, k, Time(clock.time)]
+@inline getbc(f::YZFTS, j::Int, k::Int, grid::AbstractGrid, clock, args...) = @inbounds f[1, j, k, Time(clock.time)]
+@inline getbc(f::XFTS,  i::Int, j::Int, grid::AbstractGrid, clock, args...) = @inbounds f[i, 1, 1, Time(clock.time)]
+@inline getbc(f::YFTS,  i::Int, k::Int, grid::AbstractGrid, clock, args...) = @inbounds f[1, j, 1, Time(clock.time)]
+@inline getbc(f::ZFTS,  j::Int, k::Int, grid::AbstractGrid, clock, args...) = @inbounds f[1, 1, k, Time(clock.time)]
```
Comment (navidcy) [inline src/OutputReaders/field_time_series_indexing.jl]: I'm still confused about these since they are not even symmetrical to each other...

X -> i, j
Y -> i, k
Z -> j, k

Comment (navidcy) [reply src/OutputReaders/field_time_series_indexing.jl]: The `f::YFTS` must have a typo because j only appears on the right hand side.

File: src/OutputReaders/field_time_series_indexing.jl [hunk 2]
```diff
@@ -172,21 +172,35 @@ const YFTS  = FlavorOfFTS{Nothing, <:Any, Nothing}
...
+@inline getbc(f::YZFTS, j::Int, k::Int, grid::AbstractGrid, clock, args...) = @inbounds f[1, j, k, Time(clock.time)]
+
+# Disambiguation for 1D and 0D FTS
+
+# South - north and top - bottom, only the first index is valid (the second one is either j or k)
+@inline getbc(f::XFTS, i::Int, ::Int, grid::AbstractGrid, clock, args...) = @inbounds f[i, 1, 1, Time(clock.time)]
+
+# West - east and top - bottom, this case is not really well defined since the indexes could be i and j or j and k
+# so we check which dimension of the grid is 1 and pick the corresponding index
+@inline function getbc(f::YFTS, i1::Int, i2::Int, grid::AbstractGrid, clock, args...)
```
Comment (giordano) [inline src/OutputReaders/field_time_series_indexing.jl]:
```suggestion
@inline function getbc(f::YFTS, i1::Int, i2::Int, grid::AbstractGrid, clock, args...)
```

**Review by navidcy — APPROVED** [review]
that's clean @giordano!

**Comment by giordano** [comment]
Just a minor note: I presume this is technically breaking as we're removing existing methods, right?  Although I don't think many people will miss them, since some of them were clearly broken before this PR.

**Comment by simone-silvestri** [comment]
Yeah that feature could not be used so technically it was not "supported"?
